discord/context_managers.py

```diff
@@ -53,21 +53,35 @@ class Typing:
     def __init__(self, messageable: Messageable) -> None:
         self.loop: asyncio.AbstractEventLoop = messageable._state.loop
         self.messageable: Messageable = messageable
+        self.channel: Optional[MessageableChannel] = None
 
-    async def do_typing(self) -> None:
-        try:
-            channel = self._channel
-        except AttributeError:
-            channel = await self.messageable._get_channel()
+    async def _get_channel(self) -> MessageableChannel:
+        if self.channel:
+            return self.channel
+
+        self.channel = channel = await self.messageable._get_channel()
+        return channel
+
+    async def wrapped_typer(self) -> None:
+        channel = await self._get_channel()
+        await channel._state.http.send_typing(channel.id)
+
+    def __call__(self) -> Coroutine[Any, Any, None]:
+        return self.wrapped_typer()
```

Comment:
This isn't required unless you want to support
```py
async with ctx.typing()(): ...
```
right?

Reply:
oop! yes that's correct, committing to fix this now